contentpacks/khanacademy.py

```diff
@@ -147,57 +150,32 @@ def retrieve_language_resources(version: str, sublangargs: dict, ka_domain: str,
     return LangpackResources(node_data, subtitle_data, kalite_catalog, ka_catalog)
 
 
-@cache_file
-def retrieve_subtitle_meta_data(url, path):
-
-    try:
-        response = requests.get(url)
-        response.raise_for_status()
-    except requests.HTTPError:
-        raise
-
-    content = ujson.loads(response.content)
-
-    if not content.get("objects"):
-        raise KeyError
-
-    amara_id = content["objects"][0]["id"]
-
-    with open(path, 'w') as f:
-        f.write(amara_id)
-
 
 def retrieve_subtitles(videos: list, lang=EN_LANG_CODE, force=False, threads=NUM_PROCESSES) -> dict:
     # videos => contains list of youtube ids
     """return list of youtubeids that were downloaded"""
-    lang = lang.lower()         # Amara likes lowercase codes
+    lang = lang.lower()
     def _download_subtitle_data(youtube_id):
-
         logging.info("trying to download subtitle for %s" % youtube_id)
-        request_url = "https://www.amara.org/api2/partners/videos/?format=json&video_url=http://www.youtube.com/watch?v=%s" % (
-            youtube_id
-        )
-
-        try:
-            amara_id_file = retrieve_subtitle_meta_data(request_url, filename="subtitles/meta_data/{youtube_id}".format(
-                youtube_id=youtube_id))
-            with open(amara_id_file, 'r') as f:
-                amara_id = f.read()
-            subtitle_download_uri = "https://www.amara.org/api/videos/%s/languages/%s/subtitles/?format=vtt" % (
-                amara_id, lang)
-            filename = "subtitles/{lang}/{youtube_id}.vtt".format(lang=lang, youtube_id=youtube_id)
-            subtitle_path = download_and_cache_file(subtitle_download_uri, filename=filename, ignorecache=False)
-            logging.info("subtitle path: {}".format(subtitle_path))
+        # Store the file temporarily, then rename it later since it has a wrong extension used 
+        # e.g {youtube_id}. {Lang}. vtt so we need to rename it.
+        dump_file = "%s/%s/%s" % (SUBTITLE_DIR, lang, youtube_id)
+        download_cmd = "youtube-dl --sub-format 'vtt' --sub-lang '{lang}' --write-srt -o '{path}'  --skip-download 'https://www.youtube.com/watch?v={youtube_id}'".format(
+            path=dump_file, lang=lang, youtube_id=youtube_id)
+        os.system(download_cmd)
```

Comment:
Not that important, but just so you know: Youtube-dl is actually a Python library so can be accessed directory: https://github.com/rg3/youtube-dl/blob/master/README.md#embedding-youtube-dl